src/roiextractors/extractors/tiffimagingextractors/brukertiffimagingextractor.py

```diff
@@ -315,13 +316,45 @@ def get_streams(cls, folder_path: PathType) -> dict:
         streams: dict
             The dictionary of available streams.
         """
+        channel_names = cls.get_available_channels(folder_path=folder_path)
+
         natsort = get_package(package_name="natsort", installation_instructions="pip install natsort")
-        xml_root = _parse_xml(folder_path=folder_path)
-        channel_names = [file.attrib["channelName"] for file in xml_root.findall(".//File")]
-        unique_channel_names = natsort.natsorted(set(channel_names))
+        unique_channel_names = natsort.natsorted(channel_names)
         streams = dict(channel_streams=unique_channel_names)
         return streams
 
+    @staticmethod
+    def get_available_channels(folder_path: PathType) -> set[str]:
+        """
+        Extract set of available channel names from the XML configuration file in the specified folder.
+
+        Parameters
+        ----------
+        folder_path : PathType
+            The path to the folder containing the XML configuration file. It can be either a string
+            or a Path object.
+
+        Returns
+        -------
+        Set[str]
+            A set of channel names available in the first 'Frame' element found in the XML configuration file.
+        """
+        folder_path = Path(folder_path)
+        xml_file_path = folder_path / f"{folder_path.name}.xml"
+        assert xml_file_path.is_file(), f"The XML configuration file is not found at '{folder_path}'."
+
+        channel_names = set()
+        for event, elem in ElementTree.iterparse(xml_file_path, events=("start",)):
+            if elem.tag == "Frame":
+                # Get all the sub-elements in this Frame element
+                for subelem in elem:
+                    if subelem.tag == "File":
+                        channel_names.add(subelem.attrib["channelName"])
+
+                break
+
+        return channel_names
```

Comment:
Would it make sense to switch to `lxml` as in https://github.com/catalystneuro/roiextractors/pull/346 ?

Reply:
Good idea. Do you mind if I switch it in #346 just to avoid merging conficts? All of these PR are linked.

Reply:
Yeah, sounds good. thanks!